subprojects/docs/src/snippets/tasks/incrementalBuild-incrementalBuildAdvanced/groovy/build.gradle

```diff
@@ -71,7 +71,7 @@ tasks.register('instrumentClasses2', Instrument) {
 
 // tag::inferred-task-dep-with-builtby[]
 tasks.register('instrumentClassesBuiltBy', Instrument) {
-    classFiles.from fileTree(tasks.named('compileJava').map { it.destinationDir }) {
+    classFiles.from fileTree(tasks.named('compileJava').map { it.destinationDirectory.get().getAsFile() }) {
         builtBy tasks.named('compileJava')
     }
```

Comment:
I think you'll have to look at the docs to see what this is actually trying to do.  I think we're trying to demonstrate low level workarounds using builtBy. It looks really weird to be realizing Providers to demonstrate this now.

Reply:
We're showing how to add the task dependency without an explicit `dependsOn` for a task which has multiple file-based outputs.  I agree it looks better with `flatMap` and not realizing the providers.  Changed.